test/test_ssltransport.py

```diff
@@ -116,10 +116,13 @@ def start_dummy_server(
     ) -> None:
         def socket_handler(listener: socket.socket) -> None:
             sock = listener.accept()[0]
-            with self.server_context.wrap_socket(sock, server_side=True) as ssock:
-                request = consume_socket(ssock)
-                validate_request(request)
-                ssock.send(sample_response())
+            try:
+                with self.server_context.wrap_socket(sock, server_side=True) as ssock:
+                    request = consume_socket(ssock)
+                    validate_request(request)
+                    ssock.send(sample_response())
+            except (ConnectionAbortedError, ConnectionResetError):
+                return
```

Comment:
Should we use `contextlib.suppress` here and below?

Also, in what context is it useful? It's not intuitive that this could raise an exception and not actually start the server

Reply:
You don't use `contextlib.suppress` anywhere else and I find it to be confusing compared to the regular try/catch

Reply:
OK, but in what context is it useful? It's not intuitive that this could raise an exception and not actually start the server

Reply:
previously it was as if there was a `try: .... except BaseException: return` there, this is actually just limiting the exceptions to just those. This is because effectively all exceptions in a thread are ignored by default and the filterwarings=error collects those exceptions so we need to do something about the ones that do happen

rather than add PytestUnhandledThreadExceptionWarning to the filterwarnings, as a TODO, I think it's better to do the narrower workaround here